pkg/cloud_provider/metadata/metadata.go

```diff
@@ -44,10 +48,20 @@ func NewMetadataService() (Service, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to get project: %w", err)
 	}
+	instanceID, err := metadata.InstanceID()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get instance id: %w", err)
+	}
+	internalIP, err := metadata.InternalIP()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get internal IP: %w", err)
+	}
 
```

Comment:
we can log the config details for debug

Reply:
Looks like all the logic has been wrapped in the `metadata` package already, and the config is not passed in as a parameter in this function. If there's any error, it will directly be popped up to [where we call `NewMetadataService()`](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/cmd/main.go#L86)

Reply:
Synced offline, we need to log the metadata service properties after it's successfully setup.